### PR TITLE
feat: checkbox component

### PR DESCRIPTION
**Pull request by jessywork**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Other:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/components/checkbox/src/checkbox.styles.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { CheckboxStyledProps } from './checkbox.types';
+
+export const LabelStyled = styled.label`
+  margin-left: 8px;
+  cursor: pointer;
+`;
+
+export const InputStyled = styled.input<CheckboxStyledProps>`
+  appearance: none;
+  width: 12px;
```
Comment (joaoprcosta) [inline packages/components/checkbox/src/checkbox.styles.ts:11]: We should not use 'px' but 'rem'.

**Comment by joaoprcosta** [comment]
Missing unit tests.

File: packages/components/checkbox/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export { default as checkbox } from './Checkbox';
```
Comment (joaoprcosta) [inline packages/components/checkbox/src/index.ts:1]: A component name must always be uppercase.
Also, you forgot to export CheckbocGroup.

**Review by JuniorDiasOliveira — APPROVED** [review]

File: packages/components/checkbox/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export { default as Checkbox } from './Checkbox';
```
Comment (joaoprcosta) [inline packages/components/checkbox/src/index.ts:1]: You are only allowed to export default one time in each file.

Should export like this:
```suggestion
export { Checkbox } from './Checkbox';
export { CheckboxGroup } from './CheckboxGroup';
```

**Comment by joaoprcosta** [comment]
The structure of the component is not aligned with the expected and will not generate the first version of the package. Check the button component to see an example.

File: packages/components/checkbox/src/Checkbox.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      }
+    },
+    [disabled, onChange]
+  );
+
+  return (
+    <CheckboxContainerStyled
+      backgroundColor={backgroundColor}
+    >
+      <label>
```
[the remark this answers is not in the document]
Comment (jessywork) [reply packages/components/checkbox/src/Checkbox.tsx:29]: This is the way I found to make the whole area clickable

**Review by joaoprcosta — APPROVED** [review]

**Pull request merged**